### PR TITLE
Connected electrolyzer capacity to cost models

### PR DESCRIPTION
**Pull request by elenya-grant**
# Connected electrolyzer capacity to cost models

- Added `electrolyzer_size_mw` as an input to electrolyzer cost models and used value instead of accessing electrolyzer capacity from the config.
- Changed the tech config parameter for specifying electrolyzer rating in MW to specifying the number of clusters in the electrolyzer system. Previously, if the electrolyzer rating was not a multiple of the cluster capacity, then the electrolyzer performance model would simulate the number of clusters as the rounded-up value of input electrolyzer rating divided by the cluster capacity. Now - users must specify the number of clusters which helps prevent these types of discrepancies.

The tech config for specifying electrolyzer capacity went from a format like this:
changed from being formatted like this:
```yaml
electrolyzer:
  model_inputs:
    shared_parameters:
      rating: 720
    performance_parameters:
      cluster_rating_MW: 40
```

to now being formatted like this:

```yaml
electrolyzer:
  model_inputs:
    performance_parameters:
      n_clusters: 18
      cluster_rating_MW: 40
```

## PR Checklist

- [x] `CHANGELOG.md` has been updated to describe the changes made in this PR
- [x] Documentation
  - [x] Docstrings are up-to-date
  - [x] Related `docs/` files are up-to-date, or added when necessary
  - [x] Documentation has been rebuilt successfully
  - [x] Examples have been updated
- [x] Tests pass (If not, and this is expected, please elaborate in the tests section)
- [x] PR description thoroughly describes the new feature, bug fix, etc.

## Related issues
 #192 

## Impacted areas of the software

- `h2integrate/converters/hydrogen/basic_cost_model.py`: added `electrolyzer_size_mw` as an input, this value is used instead of `config.rating` in `BasicElectrolyzerCostModel`. Removed `rating` from the config
- `h2integrate/converters/hydrogen/singlitico_cost_model.py`: added `electrolyzer_size_mw` as an input, this value is used instead of `config.rating` in `SingliticoCostModel`. Removed `rating` from the config
- `h2integrate/converters/hydrogen/wombat_model.py`: changed any calls to `self.config.rating` to `rating_from_config` which is calculated as `rating_from_config = self.config.n_clusters * self.config.cluster_rating_MW`
- `h2integrate/converters/hydrogen/test/test_wombat_model.py`: updated input dictionaries to have `n_clusters: 1` instead of `rating`.
- `h2integrate/converters/hydrogen/eco_tools_pem_electrolyzer.py`: updated `ECOElectrolyzerPerformanceModelConfig` to have inputs of `n_clusters` instead of `rating`. Relevant updates made to `ECOElectrolyzerPerformanceModel` for the change to the config, and added output of `electrolyzer_size_mw`. 
- `tests/h2integrate/test_all_examples.py`: updated test values in `test_wind_h2_opt_example` (see more information in the Test section of this PR). Also made small modifications to `examples/05_wind_h2_opt/driver_config.yaml`.
- Updates to `tech_config.yaml` files in `examples/`: for examples 1, 2, 5, 8, 10 and 12 to change how electrolyzer rating is specified.

**Updated doc files**:
- `docs/user_guide/how_to_set_up_an_analysis.ipynb`
- `docs/technology_models/wombat_electrolyzer_om.md`

## Test results, if applicable

Tests were failing in `tests/h2integrate/test_all_examples.py::test_wind_h2_opt_example`. This is expected. This example aims to optimize the electrolyzer capacity to minimize LCOH while achieving some minimum hydrogen production. The reason these tests now fail is because of changes to the CapEx and OpEx of the electrolyzer cost model (these tests were based on results that were obtained from the buggy process described in the example case outlined in #192). This example uses the `singlitico_electrolyzer_cost` as the electrolyzer cost model.

Previously, the optimized electrolyzer capacity was found to be 1515.5 MW (although this isn't what's actually being simulated since this isn't a multiple of the cluster capacity but now that's fixed too). However, the tech config specifies the electrolyzer rating as 500 MW. For a 500 MW electrolyzer the adjusted technology costs are:
- wind_adj_capex = 1340728828.1249995
- wind_adj_opex = 40221864.843749985
- elec_adj_capex = 480799091.25226986
- elec_adj_opex = 11774011.153810823
which results in a total adjusted capex of 1.82152792e09 (the value Check total adjusted CapEx compares against) and total adjusted opex of 51995875.99756081 (the value Check total adjusted OpEx compares against). This shows that even though the electrolyzer capacity has increased from 500 MW to 1515.5 MW, the costs used in the tests were calculated based on an electrolyzer capacity of 500 MW.

So, with this PR, the total adjusted CapEx and total adjusted OpEx changed because the electrolyzer costs are calculated for the input electrolyzer capacity rather than the 500 MW specified in the config. These tests failing also cause LCOE (see Issue #175) and LCOH tests to fail, and the minimum total hydrogen produced to fail since the optimal design changed.

### CODE REVIEW

File: examples/05_wind_h2_opt/driver_config.yaml [hunk 1]
```diff
@@ -8,18 +8,19 @@ driver:
...
     max_iter: 100
     solver: COBYLA
-    rhobeg: 500.
+    rhobeg: 30
     debug_print: True
 
 design_variables:
   electrolyzer:
-    electrolyzer_size_mw:
+    n_clusters:
```
[the remark this answers is not in the document]
Comment (elenya-grant) [reply examples/05_wind_h2_opt/driver_config.yaml:19]: Hasn't electrolyzer capacity always been a discrete variable (at least in the performance model)? What kind of checks do you think this may need (raising errors or something else?)?

File: examples/05_wind_h2_opt/tech_config.yaml [hunk 2]
```diff
@@ -51,7 +51,9 @@ technologies:
           resize_for_enduse: False
           size_for: 'BOL' #'BOL' (generous) or 'EOL' (conservative)
           hydrogen_dmd:
-        cluster_rating_MW: 40
+        # cluster_rating_MW: 40
+        cluster_rating_MW: 20
+        n_clusters: 18
```
[the remark this answers is not in the document]
Comment (elenya-grant) [reply examples/05_wind_h2_opt/tech_config.yaml:56]: Yes - the original electrolyzer rating and cluster rating were incompatible (it says 500 but the performance model was representative of a 520 MW system). I will update it so it starts with 25 clusters, which would give 500 MW rated capacity (with the 20 MW clusters).

File: examples/05_wind_h2_opt/driver_config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaredthomas68) [reply examples/05_wind_h2_opt/driver_config.yaml:19]: I think electrolyzer size is a continuous variable in the performance model. Using the electrolyzer repo, @camirmas  and I added the ability to automatically change the other parameters of the system to make rating fully continuous, but I don't think that capability is in the H2I repo. I do think electrolyzer rating has been treated as a continuous variable in the H2I performance model, and is certainly treated as continuous with this definition for the OpenMDAO interface.

File: h2integrate/converters/hydrogen/test/test_wombat_model.py [hunk 3]
```diff
@@ -77,14 +77,14 @@ def test_wombat_error(subtests):
...
                     "shared_parameters": {
-                        "rating": 30.0,
                         "location": "onshore",
                         "electrolyzer_capex": 1295,
                         "sizing": {
                             "resize_for_enduse": False,
                             "size_for": "BOL",
                             "hydrogen_dmd": None,
                         },
+                        "n_clusters": 0.75,
```
[the remark this answers is not in the document]
Comment (jaredthomas68) [reply h2integrate/converters/hydrogen/test/test_wombat_model.py:87]: Ah, I missed that. Got it.

File: examples/05_wind_h2_opt/driver_config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (elenya-grant) [reply examples/05_wind_h2_opt/driver_config.yaml:19]: I agree with you on that it's "treated as continuous" but I disagree on the performance model representing it as continuous (although, I should check this and I can make a separate PR with a test that verifies this). Don't these two lines (below) where we round capacities show that it isn't continuous? If cluster capacity is 40 MW, then the total rated capacity could only be (for example) 40 or 80, it can't be between 40 and 80. If you tried to run a 500 MW electrolyzer system and a 520 MW electrolyzer system in H2I, you'd get the exact same performance because each one is modeling 13 clusters rated at 40 MW (at least I'm pretty sure, again, I will make a test to check this). 
- [eco_tools_pem_electrolyzer.py](https://github.com/NREL/H2Integrate/blob/d96c2ec624ca8f4a0b9de7125907408be9f0ab89/h2integrate/converters/hydrogen/eco_tools_pem_electrolyzer.py#L104)  
- [run_PEM_clusters](https://github.com/NREL/H2Integrate/blob/d96c2ec624ca8f4a0b9de7125907408be9f0ab89/h2integrate/simulation/technologies/hydrogen/electrolysis/run_PEM_master.py#L57)

The smallest the cluster capacity could be is 1 MW, because that is what the "stacks" are modeled as. So clusters can be any multiple of 1 MW, and but stack size cannot change from 1 MW within H2I. I also think that to make it fully continuous, we have to be careful about what parameters are being adjusted to make a stack a specific capacity because if you're changing anything beyond the number of cells, you'll have a different rated capacity _and a different rated efficiency_ when you change parameters like cell active area, rated current density, etc.

In the future, I'd like to revamp the electrolyzer repo models and integrate those into H2I. For the time being, I think that specifying the number of clusters and cluster capacity would reduce the mismatches between user input and what is being modeled in the performance model compared to specifying the electrolyzer capacity and the cluster capacity. The goal of this PR is to 1) ensure the cost models and performance models are based on the same electrolyzer capacity and 2) to reduce error between what electrolyzer capacity the user has input vs what electrolyzer capacity is being modeled. Its a small step towards fixing a larger issue.

File: tests/h2integrate/test_all_examples.py [hunk 4]
```diff
@@ -186,26 +186,37 @@ def test_wind_h2_opt_example(subtests):
     model.post_process()
 
     with subtests.test("Check LCOH"):
-        assert model.prob.get_val("financials_group_1.LCOH")[0] < 4.64
+        assert model.prob.get_val("financials_group_1.LCOH")[0] < 7.8
```
[the remark this answers is not in the document]
Comment (jaredthomas68) [reply tests/h2integrate/test_all_examples.py]: Interesting, somebody (probably me) should have chosen a feasible starting point.  I think we should alter the starting point of the example so it is feasible, but that is outside the scope of this PR. Could you make an issue for it?

[the remark this answers is not in the document]
Comment (elenya-grant) [reply tests/h2integrate/test_all_examples.py]: Made an Issue, see here #200
